cmd/thanos/store.go

```diff
@@ -104,102 +106,104 @@ func runStore(
 	chunkPoolSizeBytes uint64,
 	maxSampleCount uint64,
 	maxConcurrent int,
-	component string,
+	component component.Component,
 	verbose bool,
 	syncInterval time.Duration,
 	blockSyncConcurrency int,
 	filterConf *store.FilterConfig,
 ) error {
-	{
-		confContentYaml, err := objStoreConfig.Content()
-		if err != nil {
-			return err
-		}
+	statusProber := prober.NewProber(component, logger, prometheus.WrapRegistererWithPrefix("thanos_", reg))
+
+	confContentYaml, err := objStoreConfig.Content()
+	if err != nil {
+		return err
+	}
+
+	bkt, err := client.NewBucket(logger, confContentYaml, reg, component.String())
+	if err != nil {
+		return errors.Wrap(err, "create bucket client")
+	}
 
-		bkt, err := client.NewBucket(logger, confContentYaml, reg, component)
+	// Ensure we close up everything properly.
+	defer func() {
 		if err != nil {
-			return errors.Wrap(err, "create bucket client")
+			runutil.CloseWithLogOnErr(logger, bkt, "bucket client")
```

Comment:
I think we should remove `if err !=nil`, as it is run `in defer`,  `err` will be overwritten by non related code which happens after this line.

Reply:
I'd say it was meant to close the client if the `runStore` ended up with an error. 
Otherwise it should not close the client since others are still using it. 
Maybe that's why the `if err != nil` is there and the fact err will be overwritten by the following code was the intention?

But I did not touch this code actually, just removed the unneeded code block wrapping it.